tree/ntuple/v7/src/RField.cxx

```diff
@@ -574,6 +574,8 @@ ROOT::Experimental::Detail::RFieldBase::SplitValue(const RValue & /*value*/) con
 void ROOT::Experimental::Detail::RFieldBase::Attach(
    std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> child)
 {
+   if (fState != EState::kUnconnected)
+      throw RException(R__FAIL("invalid attempt to attach subfield to already connected field"));
```

Comment:
I think the field zero is always in `kUnconnected` state; otherwise, this check would fail if the model is extended through `RNTupleModel::RUpdater`.

Consequently, perhaps we should fail when trying to connect the field zero to a page source / sink (i.e. only descendant fields can be connected)?

Reply:
Good point, thanks!